[8.11] CSPM onboarding for Azure (subscription-level) and for GCP (organization-level) (#4111)
elastic/security-docs: docs/cloud-native-security/cspm-get-started-gcp.asciidoc

. From the Elastic Security *Get started* page, click *Add integrations*.
. Search for `CSPM`, then click on the result.
. Click *Add Cloud Security Posture Management (CSPM)*.
. Under *Configure integration*, select *GCP*, then either *GCP Organization* (recommended) or *Single Account*.
. Give your integration a name that matches the purpose or team of the GCP account you want to monitor, for example, `dev-gcp-project`.

[discrete]
[[cspm-set-up-cloud-access-section-gcp]]
=== Set up cloud account access

> NOTE: To setup CSPM for a GCP project, you need admin privileges for the project.

Author: ```suggestion
NOTE: To set up CSPM for a GCP project, you need admin privileges for the project.
```

Use the following commands, after replacing `<SA_NAME>` with the name of your new service account, `<ORG_ID>` with your GCP organization's ID, and `<PROJECT_ID>` with the GCP project ID of the project where you want to provision the compute instance that will run CSPM.

Create a new service account:
```
gcloud iam service-accounts create <SA_NAME> \
    --description="Elastic agent service account for CSPM" \
    --display-name="Elastic agent service account for CSPM" \
    --project=<PROJECT_ID>
```

Assign the necessary roles to the service account:
```
gcloud organizations add-iam-policy-binding <ORG_ID> \
    --member=serviceAccount:<SA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com \
    --role=roles/cloudasset.viewer

gcloud organizations add-iam-policy-binding <ORG_ID> \
    --member=serviceAccount:<SA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com \
    --role=roles/browser
```
NOTE: The `Cloud Asset Viewer` role grants read access to cloud asset metadata. The `Browser` role grants read access to the project hierarchy.

Download the credentials JSON:
```
gcloud iam service-accounts keys create <KEY_FILE> \
    --iam-account=<SA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com
```

> Keep the credentials JSON in a secure location, you will need it later.

Reviewer: Explain that the keys will be saved to your system named after <KEY_FILE> parameter.

> Keep the credentials JSON in a secure location, you will need it later.

Reviewer 2: ```suggestion
Keep the credentials JSON in a secure location; you will need it later.
```

. Name the {agent} policy. Use a name that matches the purpose or team of the cloud account or accounts you want to monitor. For example, `dev-gcp-account`.
. Click **Save and continue**, then follow the instructions to install {agent} in your chosen GCP project.

Wait for the confirmation that {kib} received data from your new integration. Then you can click **View Assets** to see your data.

[discrete]
[[cspm-set-up-manual-gcp-project]]
== Manual authentication (GCP project)

To authenticate manually to monitor an individual GCP project, you'll need to create a new GCP service account, assign it the necessary roles, generate credentials, then provide those credentials to the CSPM integration. 

Use the following commands, after replacing `<SA_NAME>` with the name of your new service account, and `<PROJECT_ID>` with your GCP project ID.

Create a new service account:
```
gcloud iam service-accounts create <SA_NAME> \
    --description="Elastic agent service account for CSPM" \
    --display-name="Elastic agent service account for CSPM" \
    --project=<PROJECT_ID>
```

Assign the necessary roles to the service account:
```

> gcloud organizations add-iam-policy-binding <PROJECT_ID> \

Reviewer: ```suggestion
gcloud projects add-iam-policy-binding <PROJECT_ID> \
```

    --member=serviceAccount:<SA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com \
    --role=roles/cloudasset.viewer

> gcloud organizations add-iam-policy-binding <PROJECT_ID> \

Reviewer: ```suggestion
gcloud projects add-iam-policy-binding <PROJECT_ID> \
```

    --member=serviceAccount:<SA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com \
    --role=roles/browser
```
NOTE: The `Cloud Asset Viewer` role grants read access to cloud asset metadata. The `Browser` role grants read access to the project hierarchy.

Download the credentials JSON:
```
gcloud iam service-accounts keys create <KEY_FILE> \
    --iam-account=<SA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com
```

> Keep the credentials JSON in a secure location, you will need it later.

Reviewer: Same
